data/conf/llap/hive-site.xml

```diff
@@ -49,7 +49,7 @@
 <!-- the Vertex resource, so this has to be defined in order to get consistent test results -->
 <property>
   <name>hive.tez.container.size</name>
-  <value>128</value>
+  <value>256</value>
```

Comment:
+1, wanted to ask the same question.

Reply:
@abstractdog do you know about it?

Reply:
I don't know, I guess we should ask @akshat0395 about the reason for this change, right?

Reply:
After JDK9, memory use seem to increase (caused by moving from a native to pure java-based implementation[1] for zip/jar files in JDK 9. Refer [JDK-8145260](https://bugs.openjdk.org/browse/JDK-8145260)) landing test cases in OOM due to insufficient memory for the Container launched. 

Class java.util.zip.ZipFile$Source is added after JDK9, which can be seen in Heapdumps

Reply:
Yes, JDK17 requires more memory as compared to JDK8 especially due to the new module system which has been introduced as well as the [JDK-8145260](https://bugs.openjdk.org/browse/JDK-8145260) which was rightly pointed out by @Indhumathi27 . Even a lot of qtests part of our precommit tests had OOM failures when moved to jdk17 due to the same. So, this increase in tez container size is required.

Reply:
this explanation makes sense as long as it's in a tez container, however, in case of llap (this is something */llap/*.xml), this setting doesn't have a direct effect on memory consumption, because the LLAP daemon runs locally in the same surefire JVM process
are we sure that OOM issues come without this setting? isn't it possible that this change has been borrowed from another test config (we it has really been necessary)

Reply:
@abstractdog Without this change, multiple qtests running with MiniLlapCliDriver run out of memory. Yes, you are right the LLAP daemon runs locally in the same surefire JVM process, but i believe that this setting is used in allocating memory resources (portion of JVM's heap space) to the thread that runs the tez task in the same JVM.

Reply:
Just my thought, and committers could have another thought. If we agree with largere memory settings, can we extract that change as another Jira/PR? We assume the increase would not only avoid false-positive OOM but also change some test results. It is hard for a reviewer to reject the possibilities that a bunch of other changes, such as JDK or dependency update, introduced the latter problem unexpectedly

Reply:
@tanishq-chugh @okumin @abstractdog @aturoczy @deniskuzZ @Aggarwal-Raghav 
Hello. We use JDK21 to run HIVE+TEZ in our downstream environment. After adopting JDK21, I observed a significant increase in memory consumption by TEZ. For TEZ container sizes ranging from 4GB to 8GB, I had to lower the value of tez.runtime.shuffle.merge.percent from the default 0.9 to 0.75 to successfully execute the jobs. Thus, there might be additional underlying issues; the mere existence of [JDK-8145260](https://bugs.openjdk.org/browse/JDK-8145260) seems insufficient to account for this problem entirely.

Reply:
Could you please try to compile Tez against JDK 17 and check the memory consumption again?

Reply:
> Could you please try to compile Tez against JDK 17 and check the memory consumption again?

i use hive and tez both compile by jdk17,and run with jdk21